haas/api.py

```diff
@@ -534,26 +596,80 @@ def headnode_detach_network(headnode, hnic):
                             # Network Code #
                             ################
 
+@rest_call('GET', '/networks', Schema({}))
+def list_networks():
+    """Lists all networks"""
+
+    get_auth_backend().require_admin()
+
+    networks = db.session.query(model.Network).all()
+    result = {}
+    for n in networks:
+        if n.access:
+            net = {'network_id': n.network_id, 'projects': sorted([p.label for p in n.access])}
+        else:
+            net = {'network_id': n.network_id, 'projects': None}
+        result[n.label] = net
+
+    return json.dumps(result, sort_keys = True)
+
+@rest_call('GET', '/network/<network>/attachments', schema=Schema({
+    'network': basestring, Optional('project'): basestring,
+}))
+def list_network_attachments(network, project=None):
+    """Lists all the attachments from <project> for <network>
+
+    If <project> is `None`, lists all attachments for <network>
+    """
+    auth_backend = get_auth_backend()
+    network = _must_find(model.Network, network)
+
+    # Access checking.
+    if project is None:
+        # List all nodes connected to network.
+        # Only the project that owns the network should be able to list attached nodes
+        auth_backend.require_project_access(network.owner)
+    else:
```

Comment:
I think right now this else is dead code; we'd talked about adding two routes that map to this api call (one with the project and one without) but I don't see the other route anywhere. That probably still needs to be added.